### PR TITLE
Replace PUT requests with PATCH for entrypoint and garden editing pages

### LINKED ISSUE #199: Update Garden/Entrypoint editing to use PATCH endpoint instead of PUT
Right now both garden editing and entrypoint editing uses the PUT endpoint. We should switch to using PATCH to better align with the backend

### PR DESCRIPTION
**Pull request by shane-jung**

### CODE REVIEW

File: garden/src/lib/utils.ts [hunk 1]
```diff
@@ -4,3 +4,24 @@ import { twMerge } from "tailwind-merge";
...
+  Object.entries(dirtyFields).reduce((acc, [key, value]) => {
+    if (value === false) return acc;
+    if (value === true) {
+      return { ...acc, [key]: values[key] };
+    }
+    if (includesTrue(value)) {
+      return { ...acc, [key]: values[key] };
+    }
+    return acc;
+  }, {} as any);
```
Comment (WillEngler) [inline garden/src/lib/utils.ts:27]: This is nice. Glad that with the form library, we don't need too much code to extract the dirty values.

File: garden/src/components/form/garden/edit/FormFields.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        {garden?.entrypoints && garden.entrypoints.length > 0 ? (
+          <div className="grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3">
+            {garden.entrypoints.map((entrypoint: any) => (
+              <EntrypointBox key={entrypoint.doi} entrypoint={entrypoint} garden={garden} />
+            ))}
+          </div>
+        ) : (
+          <p className="text-gray-500">No Entrypoints created yet.</p>
+        )}
+      </div> */}
```
Comment (WillEngler) [inline garden/src/components/form/garden/edit/FormFields.tsx]: Any reason to keep this?

File: garden/src/components/form/garden/edit/EditGardenForm.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        garden: gardenPatchRequest,
+      });
+    },
+    [garden.doi, patchGarden],
+  );
+
+  return (
+    <form onSubmit={form.handleSubmit(onSubmit)}>
+      <Form {...form}>
+        <FormFields />
```
Comment (WillEngler) [inline garden/src/components/form/garden/edit/EditGardenForm.tsx:59]: At some point I could use a walkthrough to understand our form libraries a little better. It seems like react-hook-form is used at the "lower" level and hookform is used more at the edges for validation against the zod schema. Is that right at a high level or am I way off base?

**Review by WillEngler — APPROVED** [review]
This looks great, thanks!

File: garden/src/components/form/garden/edit/EditGardenForm.tsx (hunk 3 above)
Comment (shane-jung) [reply garden/src/components/form/garden/edit/EditGardenForm.tsx:59]: Yeah, the way I understand it, hookform supplies different validators (like Zod). I think react-hook-form has its own built-in validator, but Zod is pretty clean/intuitive so I've always used that instead

React hook form supplies all of the nice form controls you'd want, removing the hassle of having a bunch of useState/useEffect for fields/errors. It keeps track of registered fields, errors, other form state, and when the form is submitted, uses the validator to make sure the fields match the expected schema. Really nice for things 'field Arrays' ([check out the "Associated Materials" section of entrypoint editing](https://github.com/Garden-AI/garden-frontend/blob/staging/garden/src/components/form/entrypoint/edit/AssociatedMaterialsSection.tsx)), which are fields that are arrays that can grow/shrink in size

File: garden/src/components/form/garden/edit/FormFields.tsx (hunk 2 above)
Comment (shane-jung) [reply garden/src/components/form/garden/edit/FormFields.tsx]: Hmm, I may revisit this now that you point this out. I think may be a good place to add the ability to attach/remove entrypoints to a garden. I'd forgotten I'd commented this out

**Pull request merged**